dashboard/app/controllers/registrations_controller.rb

```diff
@@ -46,6 +51,8 @@ def sign_up_params
       if params[:user_type] == "teacher"
         params[:email_preference_opt_in_required] = true
         params[:email_preference_request_ip] = request.env['REMOTE_ADDR']
+        params[:email_preference_source] = EmailPreference::ACCOUNT_SIGN_UP
+        params[:email_preference_form_kind] = "0"
```

Comment:
Probably out of scope for this PR, but out of curiosity, why is this `"0"`? Wouldn't it be more clear to use `nil` to represent no form kind?

Reply:
In lieu of a formal versioning strategy, I'm marking all initial versions of these strings as "0", with the idea that we can bump this each time we update the string that the user sees.  (We need to track which string we present to the user when they opt-in, and I didn't want to store a literal string for every entry.)

Reply:
That's also why I've left `form_kind` as an argument to the new `email_preference_params` helper - I expect them to change independently in the future.

Reply:
Ok, that makes sense, but what does it represent? 

I assumed this was the same as the [contact_rollups form.kind](https://github.com/code-dot-org/code-dot-org/blob/11421a96fb574a40db5e890e7c3a0a6421400cd7/lib/cdo/contact_rollups.rb#L753), but now I'm not sure. 

It's probably worth adding some documentation to the `EmailPreference` model, as a followup.